lms/security.py

```diff
@@ -177,7 +177,9 @@ def identity(self, request):
         if lti_user is None:
             return Identity("", [])
 
-        permissions = [Permissions.LTI_LAUNCH_ASSIGNMENT, Permissions.API]
+        permissions = []
+        if lti_user.is_learner or lti_user.is_instructor or lti_user.is_admin:
+            permissions = [Permissions.LTI_LAUNCH_ASSIGNMENT, Permissions.API]
```

Comment:
Only add the permission if we can explicitly recognize you as learner, instructor or admin.